compiler/const-manipulations.h

```diff
@@ -217,6 +231,11 @@ struct CheckConstWithDefines final
     return false;
   }
 
+  bool on_func_call(VertexAdaptor<op_func_call> v) override {
+    // Allows use objects in const context
+    return v->get_string() == "__construct";
```

Comment:
I suppose, it's `return v->extra_type == op_ex_constructor_call;`, comparing strings is strange